### PR TITLE
[#2445] Provider draft status

### LINKED ISSUE #2445: Draft status for providers
Draft status should be added to providers. Providers with draft status should not be visible for the user in the front office.

As per usual integration with PC (rest / jms) should be added.

This issue do not contain adding visual representation for draft status in backoffice (this is a scope of #2468).

### PR DESCRIPTION
**Pull request by kmarszalek**
- [x] view privacy for draft provider
- [x] do not show draft provider in providers page
- [x] home page, providers section
- [x] starts section for providers
- [x] resource list with draft providers on nondraft resources
- [x] no draft provider in filters in MP
- [x] no draft provider in autocomplete in MP
- [x] edit resource form with no draft providers
- [x] home page - providers section doesn't show deleted providers (counter also)
- [x] execution stats doesn't show deleted or drafted providers

Closes #2445 #2471

### CODE REVIEW

**Comment by Marcelinna** [comment]
image
image

make sure that  provider with status draft aren't shown in this summary

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +37,7 @@ Please view this file on the master branch, on stable branches it's out of date.
 - Fix RWD CSS Styles (@jarekzet)
 - Fix Offer CSS Styles on Resource view (@jarekzet)
 - Fix redirect from MP to Resource dashboard in PC (@kmarszalek)
+- Fix redirect from MP to Resource dashboard in PC (@kmarszalek)
```
Comment (kmarszalek) [inline CHANGELOG.md:40]: will delete this line after successful github tests

File: app/models/provider.rb [hunk 2]
```diff
@@ -11,15 +11,15 @@ class Provider < ApplicationRecord
 
   searchkick word_middle: [:provider_name]
 
-  STATUSES = { published: "published", deleted: "deleted" }.freeze
+  STATUSES = { published: "published", deleted: "deleted", draft: "draft" }.freeze
```
Comment (jswk) [inline app/models/provider.rb:14]: Maybe change the order to better map provider's lifecycle?
```ruby
   STATUSES = { draft: "draft", published: "published", deleted: "deleted" }.freeze
```

File: spec/services/provider/pc_create_or_update_spec.rb [hunk 3]
```diff
@@ -35,6 +35,7 @@
     expect do
       described_class.new(
         create(:jms_provider_response, eid: "new.provider", name: "Supper new name for updated  provider"),
+        true,
```
Comment (jswk) [inline spec/services/provider/pc_create_or_update_spec.rb:38]: Please create an analogous test for `active: false`.

**Review by jswk — CHANGES_REQUESTED** [review]

**Review by jswk — APPROVED** [review]

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**